app/controllers/codelijsten/edit.js

```diff
@@ -26,183 +25,152 @@ export default class CodelijstenEditController extends Controller {
+    this.schemeDescription = event.target.value.trim() ?? '';
+    this.schemeDescriptionErrorMessage = null;
+    this.validateSchemeDescription
+      .perform()
+      .then(() => this.setSaveButtonState());
+  }
+
+  validateSchemeDescription = restartableTask(async () => {
+    if (!this.schemeDescription || this.schemeDescription.trim() == '') {
+      this.schemeDescriptionErrorMessage = this.intl.t(
+        'constraints.mandatoryField'
+      );
+      return;
     }
 
-    this.codelistDescription = newDescription.trim();
-
-    if (this.codelistDescription.length > DESCRIPTION_INPUT_CHAR_LIMIT) {
-      this.descriptionErrorMessage = this.intl.t(
+    if (this.schemeDescription.length > DESCRIPTION_INPUT_CHAR_LIMIT) {
+      this.schemeDescriptionErrorMessage = this.intl.t(
         'constraints.maxCharactersReachedWithCount',
         {
-          count: this.codelistDescription.length,
+          count: this.schemeDescription.length,
           maxCount: DESCRIPTION_INPUT_CHAR_LIMIT,
         }
       );
+      return;
     }
 
-    this.setIsSaveButtonDisabled();
-  }
+    this.schemeDescriptionErrorMessage = null;
+  });
 
   @action
-  async handleNameChange(event) {
-    const newName = event.target.value;
-    this.nameErrorMessage = null;
-
-    if (!newName || newName.trim() == '') {
-      this.nameErrorMessage = this.intl.t('constraints.mandatoryField');
-    }
-
-    this.codelistName = newName.trim();
-    this.isDuplicateName = await isDuplicateConceptSchemeName(
-      this.conceptScheme,
-      this.codelistName,
-      this.store
-    );
-
-    if (this.codelistName !== '' && this.isDuplicateName) {
-      this.nameErrorMessage = this.intl.t('constraints.duplicateName');
-    }
-
-    if (this.codelistName.length > NAME_INPUT_CHAR_LIMIT) {
-      this.nameErrorMessage = this.descriptionErrorMessage = this.intl.t(
-        'constraints.maconstraints.maxCharactersReached'
-      );
-    }
-
-    this.setIsSaveButtonDisabled();
-  }
+  updateConcept(concept, event) {
+    const label = event.target.value.trim() ?? '';
+    const foundConcept = this.conceptList.find((c) => c.id == concept.id);
+    const indexOfConcept = this.conceptList.indexOf(foundConcept);
+    this.conceptList[indexOfConcept].label = label;
 
-  @action
-  handleConceptChange(concept, event) {
-    if (event.target && event.target.value.trim() == '') {
+    if (label == '') {
       showErrorToasterMessage(
         this.toaster,
         this.intl.t('constraints.optionCannotBeEmpty', { label: concept.label })
       );
+      return;
     }
 
-    const foundConcept = this.concepts.find((c) => c.id == concept.id);
-    this.concepts[this.concepts.indexOf(foundConcept)].label =
-      event.target.value.trim();
-
-    this.setIsSaveButtonDisabled();
+    this.setSaveButtonState();
   }
 
+  saveUnsavedChanges = restartableTask(async () => {
+    await this.save();
+    this.isSaveModalOpen = false;
+    this.goToNextRoute();
```

Comment:
This feels like _mAgIc_ what is the next route? Is it clear that the next route has been set in the codelijsten.edit route?